src/Database/LSMTree/Internal.hs

```diff
@@ -1409,9 +1405,34 @@ deriving stock instance Read BloomFilterAlloc
 defaultBloomFilterAlloc :: BloomFilterAlloc
 defaultBloomFilterAlloc = AllocFixed 10
 
-bloomFilterAllocForLevel :: BloomFilterAlloc -> LevelNo -> RunBloomFilterAlloc
-bloomFilterAllocForLevel (AllocFixed n) _        = RunAllocFixed n
-bloomFilterAllocForLevel (AllocRequestFPR fpr) _ = RunAllocRequestFPR fpr
+bloomFilterAllocForLevel :: TableConfig -> LevelNo -> RunBloomFilterAlloc
+bloomFilterAllocForLevel conf (LevelNo l) =
+    assert (l > 0) $
+    case confBloomFilterAlloc conf of
+      AllocFixed n -> RunAllocFixed n
+      AllocRequestFPR fpr -> RunAllocRequestFPR fpr
+      AllocMonkey totalBits (NumEntries n) ->
+        let !sr = sizeRatioInt (confSizeRatio conf)
+            !m = case confWriteBufferAlloc conf of
+                    AllocNumEntries (NumEntries x) -> x
+            !levelCount = Monkey.numLevels (fromIntegral n) (fromIntegral m) (fromIntegral sr)
+            !allocPerLevel = Monkey.monkeyBits
+                                (fromIntegralChecked totalBits)
+                                (fromIntegralChecked n)
+                                (fromIntegralChecked sr)
+                                levelCount
```

Comment:
`allocPerLevel` is fixed for a given conf, so it could be re-used. Not sure it makes a difference, though, as the cost is probably insignificant compared to performing the actual merge.

Reply:
Yup, though this seems to be the simpler option and I don't think recomputing this is a bottleneck